jobserv/trigger.py

```diff
@@ -70,6 +76,14 @@ def trigger_runs(
         raise ApiError(500, str(e) + "\n" + traceback.format_exc())
 
 
+def _fail_run(run, reason):
+    run.set_status(BuildStatus.FAILED)
+    storage = Storage()
+    with storage.console_logfd(run, "a") as f:
+        f.write(reason)
+    storage.copy_log(run)
```

Comment:
I guess this weird code is copied from elsewhere, right?
Do you want to reuse it instead?

Reply:
its kind of a pain to refactor it right now. i'll wait till it gets a little worse